### PR TITLE
fixes for `deno` compatibility

### PR DESCRIPTION
**Pull request by mash-graz**
Just a first draft PR to feed the CI pipeline, because local testing doesn't work on my machine.

Fixes: #385

Warning! -- there are important parts missing/untested!

* `validateHeaderName` and `validateHeaderValue` are temporary replaced by dummy entries.

  This should be better fixed on the `deno` side (see: https://github.com/denoland/deno/issues/22614).

 * The `worker_thread.unref()` case is still uttery unsolved and just temporary removed.

### CODE REVIEW

File: packages/preview2-shim/lib/synckit/index.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ export function createSyncFn(workerPath, debug, callbackHandler) {
     }
     return result;
   };
-  worker.unref();
+  // worker.unref();
```
Comment (guybedford) [inline packages/preview2-shim/lib/synckit/index.js:90]:
```suggestion
  if (worker.unref) worker.unref();
```

File: packages/preview2-shim/lib/nodejs/http.js [hunk 2]
```diff
@@ -21,9 +21,12 @@ import {
   registerDispose,
   registerIncomingHttpHandler,
 } from "../io/worker-io.js";
-import { validateHeaderName, validateHeaderValue } from "node:http";
+// import { validateHeaderName, validateHeaderValue } from "node:http";
```
Comment (guybedford) [inline packages/preview2-shim/lib/nodejs/http.js:24]:
```suggestion
import * as http from "node:http";
const { validateHeaderName = () => {}, validateHeaderValue = () => {} } = http;
```

**Review by guybedford — APPROVED** [review]
I'd be happy to land this with the suggested changes.

File: packages/preview2-shim/lib/synckit/index.js (hunk 1 above)
Comment (mash-graz) [reply packages/preview2-shim/lib/synckit/index.js:90]: yes -- that's a much nicer solution! :)

Nevertheless, I think, it's not enough!

The timeouts in the CI pipeline are most likely caused by hanging processes affected by this particular change resp. broken termination of workers. :(

We have to check for active event listeners or timers...

File: packages/preview2-shim/lib/nodejs/http.js (hunk 2 above)
Comment (mash-graz) [reply packages/preview2-shim/lib/nodejs/http.js]: That's also a very nice improvement!

I just committed both changes, but it still needs a lot more of additional testing and rework.

**Comment by guybedford** [comment]
Thanks for working on this. Let's at least land for now, we can always improve later.

**Comment by mash-graz** [comment]
> Thanks for working on this. Let's at least land for now, we can always improve later.

Yes -- that's a good interim solution, because the merged changes seem to do no harm to already existing use cases and test routines.

I just waited, because I could isolate the non-termination issue as another `deno` flaw (see:  https://github.com/denoland/deno/issues/22629) and I'm still searching a working solution to reuse the existing `mocha` tests also for `deno` unit checks (That's rather tricky, because it's very hard to create working `import_map`-redirections in mixed `node`/`deno`-workflows). But it's perfectly fine for me to send you these additional improvements in new PRs.

Thanks for all your help! It's really a pleasure to collaborate with you!

**Pull request merged**